[gauravuj] iP

gauravuj opened the pull request:
# Sophia: the Next Generation Chatbot

> "Quality is much better than quantity. One home run is much better than two doubles." -Steve Jobs ([source](https://leaders.com/articles/leaders-stories/steve-jobs-quotes/))

#### Sophia helps to make things easier for you. It can remember anything. And it is,

1. **FREE**
2. _SUPER FAST_
3. **Cute & Friendly** (yay!)

## What can Sopia do for you?

- remember your ~~boring~~ **_amazing_** tasks 💯 
- create a **user profile** for you 😄 
- make _todos_, _events_, and _deadline_ tasks ✅ 
- have a **chat** about your day 👨🏻‍🚀 

## How to give commands to Sophia?

1.  First, `enter your name.`
2. Now add something you want to do today, for example `todo go to gym`
3. If you want, create a deadline task `deadline submit assignment /by Sunday 23:59`
4. Spice up with some events by `event Bruno Mars concert /from 2024-04-05 20:00 /to 2024-04-05 10:00`
5. Finally, see all tasks in the list by using `list` command!

## Step-by-step: How to launch the application?

- [X] Navigate to the [Latest Release](https://github.com/gauravuj/ip/releases)
- [ ] Click on **Duke v1.0**
- [ ] Then, download the `IP.jar` into a folder
- [ ] Open **Terminal** and navigate to the folder using `cd {filepath}`
- [ ] Finally, run the program in your Terminal using `java -jar IP.jar` command!

## List of Commands
| Commands      | Description |
| :---        |    :----   |
| `list`    | prints out all the tasks you have created and saved       |
| `mark`   | marks the task as complete       |
| `unmark`   | unmarks the task as incomplete       |
| `todo`   | creates a todo task       |
| `event`   | creates an event task that include **from** and **to**       |
| `deadline`   | creates a deadline task that include the **deadline date**       |
| `find`   | find similar tasks with given **keyword**       |


## Wanna add something more? 
- Try cloning the repo or create a pull request with your suggestions. 
- Review the code for inspiration, if you like!

Here's a preview of the `main` method:
```java
public class Duke {
   public static void main(String[] args) {
      new Duke().run();
   }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

gauravuj committed:
initial development with rename, greet, and exit

gauravuj committed:
initial development with echo

gauravuj committed:
initial development with add list

gauravuj committed:
initial development with mark, unmark

gauravuj committed:
initial development with todos, events and deadlines

gauravuj committed:
initial development with textui-testing

gauravuj committed:
initial development with exceptions

gauravuj committed:
initial development with delete task

gauravuj committed:
initial development with A-enums

gauravuj committed:
development stage-2 with save feature to user dir

gauravuj committed:
development stage-2 with proper handling of fileLoad and fileSave error

gauravuj committed:
development stage-2 with Save to file feature

gauravuj committed:
development stage-2 with understanding of dates and times

gauravuj committed:
development stage-2 addition of Dates and Times feature

gauravuj committed:
create more oop classes

gauravuj committed:
create more oop classes

gauravuj committed:
divide into different packages

gauravuj committed:
divide into required packages

gauravuj committed:
add gradle support

gauravuj committed:
test with junit tests

gauravuj committed:
create jar file and release

gauravuj committed:
create javadocs for all classes/methods

gauravuj committed:
create javadoc for methods and classes

gauravuj committed:
modify Java coding standard in classes

gauravuj committed:
resolve merge conflicts

gauravuj committed:
sync changes from master

gauravuj committed:
add find feature to search tasks using a keyword

gauravuj committed:
add find feature to search tasks using a keyword

radeon2525 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,166 @@
+        }
+    }
+
+    private void handleMarkCommand(String[] userMessage, TaskManager taskManager, String input) {
```
Add the javaDocs relating to handleMarkCommand

radeon2525 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,166 @@
+        taskManager.displayTask(input);
+    }
+
+    private void handleUnmarkCommand(String[] userMessage, TaskManager taskManager, String input) {
```
Add the javaDocs relating to handleUnmarkCommand

radeon2525 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,166 @@
+        taskManager.displayTask(input);
+    }
+
+    private void handleDeleteCommand(String[] userMessage, TaskManager taskManager, String input) {
```
Add the javaDocs relating to handleDeleteCommand

radeon2525 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,166 @@
+        taskManager.deleteTask(Integer.parseInt(userMessage[1]) - 1);
+    }
+
+    private void handleFindCommand(String[] userMessage, TaskManager taskManager) {
```
Add the javaDocs relating to handleFindCommand

radeon2525 reviewed (commented):
Overall, the code is very well written with not much coding standard violation and the classes are also properly structured.

gauravuj committed:
checkstyle for duke

gauravuj committed:
checkstyle for git

menatahari commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,97 @@
+ * for managing deadlines.
+ */
+public class Deadline extends Task {
+    private LocalDateTime by;
```
Should by be a more intuitive and specific name? Something like byDate or byTime. Perhaps you can think of some other alternative name

menatahari commented on src/main/java/duke/task/FileSaveException.java:
```diff
@@ -0,0 +1,14 @@
+        super(message, cause);
+    }
+
+    public static class SaveError extends FileSaveException {
```
I find the class name a bit ambiguous as it can be interpreted as a verb. Perhaps you can add a word to make it less ambiguous, maybe something like FileSaveError?

menatahari commented on src/main/java/duke/conversation/Conversation.java:
```diff
@@ -0,0 +1,89 @@
+     *
+     * @param username The optional username for personalized dialogues.
+     */
+    public void initializeDialogues(String username) {
```
I like how descriptive and detailed your method namings are. Good work!

menatahari commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,166 @@
+     *
+     * @param input The user input to be parsed and executed.
+     */
+    public void parse(String input) {
```
Perhaps the method name can be more detailed? Maybe something like parseInput.

menatahari reviewed (commented):
Your overall method and variable namings are really detailed and specific. I just added some minor suggestions you can consider.
Really great work! I learned a lot from seeing your code.

gauravuj committed:
stage-1 of javafx ui development

gauravuj committed:
stage-2 of javafx development with errors fixed

gauravuj committed:
final-stage development of javafx ui development

gauravuj committed:
implement gui using javafx and scene builder

gauravuj committed:
changes to few issues such as label background

gauravuj committed:
issues fixed

gauravuj committed:
integrate OpenAI to make it more interactive

gauravuj committed:
Improve assertion checks in TaskDisplay class

The current situation is that assertion checks in the TaskDisplay class are incorrect,
leading to potential misbehavior or missed errors during runtime.

It needs to change because correct assertion checks are crucial for ensuring the reliability
and robustness of the codebase, especially in handling user input and command processing.

Corrected the assertion conditions in the TaskDisplay class to accurately reflect the intended
validation logic. This ensures that potential issues related to invalid command syntax or missing
arguments are appropriately caught during runtime.

The assertion checks are now aligned with the actual conditions that need to be validated,
enhancing the code's readability and maintainability. By ensuring accurate assertion checks,
potential runtime errors can be identified and addressed more effectively during development and testing.

gauravuj committed:
improve code quality through division of command handlers

gauravuj committed:
Merge pull request #1 from gauravuj/branch-A-Assertions

Branch A-Assertions

gauravuj committed:
Merge branch 'master' into branch-A-CodeQuality

gauravuj committed:
Merge pull request #2 from gauravuj/branch-A-CodeQuality

Branch A-CodeQuality

gauravuj committed:
implement a simple syntax for accessing commands

gauravuj committed:
implement friendlier syntax with defining shortcuts method

gauravuj committed:
friendlier syntax implement with user define method

gauravuj committed:
improve user interface stage-0

gauravuj committed:
implement user interface stage-1

gauravuj committed:
implement user interface final-stage

gauravuj committed:
Merge pull request #4 from gauravuj/branch-A-BetterGui

Improved GUI Quality

gauravuj committed:
Sophia's personality set to being friendly and goofy

gauravuj committed:
remove emoji parser

gauravuj committed:
Merge pull request #5 from gauravuj/branch-A-Personality

Improved Sophia's Personality

gauravuj committed:
edit the AI.md file and add some javadocs

gauravuj committed:
Merge pull request #6 from gauravuj/branch-A-AiAssisted

Add AI.md file with reflections and usage

gauravuj committed:
Merge branch 'master' into branch-Open-AI

gauravuj committed:
Merge pull request #7 from gauravuj/branch-Open-AI

Integrate OpenAI to make it more interactive

gauravuj committed:
update README and set up Page

gauravuj committed:
update README

gauravuj committed:
Update README.md

gauravuj committed:
Update README.md

gauravuj committed:
Update README.md

gauravuj committed:
Update README.md
